logstash-core/lib/logstash/plugin.rb

```diff
@@ -8,8 +8,26 @@
 require "logstash/plugins/registry"
 
 class LogStash::Plugin
+
+
+  class Timer
```

Comment:
I wonder if we can make [TimedExecution](https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/instrument/metric.rb#L84) and this `Timer` class use the same code? Low priority, since the code is very simple.